src/main/java/org/mockito/internal/util/DefaultMockingDetails.java

```diff
@@ -36,17 +38,25 @@ public boolean isSpy(){
 
     @Override
     public Collection<Invocation> getInvocations() {
-        return getMockHandler(toInspect).getInvocationContainer().getInvocations();
+        return mockHandler().getInvocationContainer().getInvocations();
     }
 
     @Override
-    public Class<?> getMockedType() {
-        return getMockHandler(toInspect).getMockSettings().getTypeToMock();
+    public MockCreationSettings<?> getMockCreationSettings() {
+        return mockHandler().getMockSettings();
     }
 
-    @Override
-    public Set<Class<?>> getExtraInterfaces() {
-        return getMockHandler(toInspect).getMockSettings().getExtraInterfaces();
+    private InternalMockHandler<Object> mockHandler() {
+        assertGoodMock();
```

Comment:
We cannot have them on constructor at the moment because isMock / isSpy methods permit null / non-mock arguments. Options:

1. separate isMock / isSpy API somewhere else
2. return empty / null result from getInvocations / getCreationSettings - then we don't need any exception
3. leave things as they are (merge PR :D)
4. ?

My thinking process

1 - API discoverability suffers
2 - poor API, no distinction between non-mock and mock without invocations, null handling required for the client code
3 - late exception not ideal but acceptable IMO

Good feedback!